src/Api/Serializer/XmlBody.php

```diff
@@ -38,7 +38,7 @@ public function build(Shape $shape, array $args)
         $xml = new XMLWriter();
         $xml->openMemory();
         $xml->startDocument('1.0', 'UTF-8');
-        $this->format($shape, $shape['locationName'], $args, $xml);
+        $this->format($shape, $shape['locationName'] ?: $shape['name'], $args, $xml);
```

Comment:
We don't have specific unit tests for XmlBody; this would be covered by the protocol test cases.